Add Phi4 test and fix regex parsing.

shoumikhin opened the pull request:
Summary:
- Add Phi4 test
- Handle the invert flag properly
- Skip regex lookahead since it's not supported by RE2
- Catch std::exception instead of narrow json parser errors
- Make the special tokens arg optional in ObjC/Swift API

Differential Revision: D83641294

larryliu0820 reviewed (approved):
Review automatically exported from Phabricator review in Meta.

shoumikhin committed:
Add Phi4 test and fix regex parsing. (#14716)

Summary:

X-link: https://github.com/meta-pytorch/tokenizers/pull/130

- Add Phi4 test
- Handle the invert flag properly
- Skip regex lookahead since it's not supported by RE2
- Catch std::exception instead of narrow json parser errors
- Make the special tokens arg optional in ObjC/Swift API

Reviewed By: larryliu0820, metascroy

Differential Revision: D83641294

shoumikhin commented:
@pytorchbot cherry-pick --onto release/1.0 -c fixnewfeature
